mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerBaseTest.java

```diff
@@ -112,113 +109,73 @@
 
   // before variables
   protected static TestHiveMetastore metastore;
+  private static TestHiveShell shell;
 
   private TestTables testTables;
 
   public abstract TestTables testTables(Configuration conf, TemporaryFolder tmp) throws IOException;
 
+  @Parameters(name = "fileFormat={0}")
+  public static Iterable<FileFormat> fileFormats() {
+    return ImmutableList.of(FileFormat.PARQUET, FileFormat.ORC, FileFormat.AVRO);
+  }
+
+  @Parameter
+  public FileFormat fileFormat;
 
   @BeforeClass
   public static void beforeClass() {
     metastore = new TestHiveMetastore();
     // We need to use increased pool size in these tests. See: #1620
     metastore.start(METASTORE_POOL_SIZE);
+    shell = new TestHiveShell();
+
+    String metastoreUris = metastore.hiveConf().getVar(HiveConf.ConfVars.METASTOREURIS);
+    shell.setHiveConfValue(HiveConf.ConfVars.METASTOREURIS.varname, metastoreUris);
+    String metastoreWarehouse = metastore.hiveConf().getVar(HiveConf.ConfVars.METASTOREWAREHOUSE);
+    shell.setHiveConfValue(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, metastoreWarehouse);
+    shell.setHiveConfValue("hive.notification.event.poll.interval", "-1");
+
+    shell.start();
```

Comment:
Wouldn't it make sense to embed the `TestHiveMetastore` in `TestHiveShell` so that it is always available? Or is there a case where we use `TestHiveShell` without `TestHiveMetastore`?

Reply:
That's right, `TestHiveShell` always needs an external metastore instance, so yes I agree it makes sense to embed the `TestHiveMetastore`